crates/bevy_render/src/render_asset.rs

```diff
@@ -38,6 +39,12 @@ pub trait RenderAsset: Asset + Clone {
     /// Whether or not to unload the asset after extracting it to the render world.
     fn asset_usage(&self) -> RenderAssetUsages;
 
+    /// Size of the data the asset will upload to the gpu. Specifying a return value
+    /// will allow the asset to be throttled via [`RenderAssetBytesPerFrame`].
+    fn byte_len(&self) -> Option<usize> {
```

Comment:
```suggestion
    fn byte_len(source_asset: &Self::SourceAsset) -> Option<usize> {
```

This should make it compatible with main, just need to update the implementations.